cassandane/Cassandane/Cyrus/JMAPComposite.pm

```diff
@@ -1,386 +0,0 @@
-#!/usr/bin/perl
-#
-#  Copyright (c) 2017 FastMail Pty Ltd  All rights reserved.
-#
-#  Redistribution and use in source and binary forms, with or without
-#  modification, are permitted provided that the following conditions
-#  are met:
-#
-#  1. Redistributions of source code must retain the above copyright
-#     notice, this list of conditions and the following disclaimer.
-#
-#  2. Redistributions in binary form must reproduce the above copyright
-#     notice, this list of conditions and the following disclaimer in
-#     the documentation and/or other materials provided with the
-#     distribution.
-#
-#  3. The name "Fastmail Pty Ltd" must not be used to
-#     endorse or promote products derived from this software without
-#     prior written permission. For permission or any legal
-#     details, please contact
-#      FastMail Pty Ltd
-#      PO Box 234
-#      Collins St West 8007
-#      Victoria
-#      Australia
-#
-#  4. Redistributions of any form whatsoever must retain the following
-#     acknowledgment:
-#     "This product includes software developed by Fastmail Pty. Ltd."
-#
-#  FASTMAIL PTY LTD DISCLAIMS ALL WARRANTIES WITH REGARD TO
-#  THIS SOFTWARE, INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
-#  AND FITNESS, IN NO EVENT SHALL OPERA SOFTWARE AUSTRALIA BE LIABLE
-#  FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
-#  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN
-#  AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
-#  OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
-#
-
-package Cassandane::Cyrus::JMAPComposite;
-use strict;
-use warnings;
-use DateTime;
-use JSON::XS;
-use Net::CalDAVTalk 0.09;
-use Net::CardDAVTalk 0.03;
-use Mail::JMAPTalk 0.13;
-use Data::Dumper;
-use Storable 'dclone';
-use MIME::Base64 qw(encode_base64);
-use Cwd qw(abs_path getcwd);
-use URI;
-use URI::Escape;
-
-use lib '.';
-use base qw(Cassandane::Cyrus::TestCase);
-use Cassandane::Util::Log;
-
-use charnames ':full';
-
-sub new
```

Comment:
Does the JMAPComposite even need a `new` function? I haven't dug into the recent Cassandane refactor, but don't we still have distinct JMAPTalk instances per JMAPEmail, JMAPCalendars, ... test suites?

Reply:
I think I should have first had my morning coffee before doing code review. Sorry for the noise.